reflex/components/graphing/recharts/utils.py

```diff
@@ -0,0 +1,389 @@
+    pandas.DataFrame
+
+    """
+    if is_type(data,"pandas.core.frame.DataFrame"):
+        return data.copy() if ensure_copy else cast(DataFrame, data)
+
+
+    if is_type(data, "numpy.ndarray"):
+        if len(data.shape) == 0:
+            return DataFrame([])
+        return DataFrame(data)
+
+    # Try to convert to pandas.DataFrame. This will raise an error is df is not
+    # compatible with the pandas.DataFrame constructor.
+    try:
+        return DataFrame(data)
+
+    except ValueError:
+        if isinstance(data, dict):
+            with contextlib.suppress(ValueError):
+                # Try to use index orient as back-up to support key-value dicts
+                return DataFrame.from_dict(data, orient="index")
+        raise TypeError(f"Unable to convert object of type `{type(data)}` to `pandas.DataFrame`.")
+        
+
+
+def _parse_x_column(df: pd.DataFrame, x_from_user: str | None) -> str | None:
+    if x_from_user is None:
+        return None
+
+    elif isinstance(x_from_user, str):
+        if x_from_user not in df.columns:
+            raise ValueError(f"{x_from_user} not in columns in data.")
+
+        return x_from_user
+
+    else:
+        raise TypeError(
+            f"x parameter should be a column name (str) or None to use the dataframe's index. Value given: {x_from_user}, (type {type(x_from_user)})"
+        )
+
+
+def is_sequence(seq: Any) -> bool:
+    """True if input looks like a sequence."""
+    if isinstance(seq, str):
+        return False
+    try:
+        len(seq)
+    except Exception:
+        return False
+    return True
+
+
+def _parse_y_columns(
+    df: pd.DataFrame,
+    y_from_user: str | Sequence[str] | None,
+    x_column: str | None,
+) -> List[str]:
+    y_column_list: List[str] = []
+
+    if y_from_user is None:
+        y_column_list = list(df.columns)
+
+    elif isinstance(y_from_user, str):
+        y_column_list = [y_from_user]
+
+    # this allows y to take in a list of columns, come back to this
+    elif is_sequence(y_from_user):
+        y_column_list = list(str(col) for col in y_from_user)
+
+    else:
+        raise TypeError(
+            f"y parameter should be a column name (str) or list thereof. Value given: {y_from_user} (type {type(y_from_user)})"
+        )
+
+    for col in y_column_list:
+        if col not in df.columns:
+            raise ValueError(f"{col} not in columns in data.")
+
+    # y_column_list should only include x_column when user explicitly asked for it.
+    if x_column in y_column_list and (not y_from_user or x_column not in y_from_user):
+        y_column_list.remove(x_column)
+
+    return y_column_list
+
+
+def _maybe_reset_index_in_place(
+    df: pd.DataFrame, x_column: str | None, y_column_list: List[str]
+) -> str | None:
+    if x_column is None and len(y_column_list) > 0:
+        if df.index.name is None:
+            # Pick column name that is unlikely to collide with user-given names.
+            x_column = "index--p5bJXXpQgv"
```

Comment:
seems like this should be dynamic, perhaps using `rx.vars.get_unique_variable_name`, or at least pulled out to a top level variable for reuse without repetition... however, if this will end up in the compiled code, it's probably better to have a stable value that doesn't change on every compile. In other parts of the code, i hash the output object/code/whatever and use the hash as the identifier.